lib/practice_exercises.rb

```diff
@@ -1,13 +1,31 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)?
+# Space Complexity: O(1)?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string == ""
+
+  letter_counter = 0
+  new_string = string.downcase.gsub(/[^A-Za-z0-9]\s*/, '')
+  new_string_array = new_string.chars
+  length = new_string_array.length
+
+  while letter_counter < length
+    if new_string_array[0] == new_string_array[new_string_array.length - 1]
```

Comment:
I think you need to use `letter_counter` in the indexes here.

```suggestion
    if new_string_array[letter_counter] == new_string_array[new_string_array.length - 1 - letter_counter]
```

You also need to think, do you return true if two letters match?  That doesn't quite work as "tott" would return true and it shouldn't.